trio/_core/_io_kqueue.py

```diff
@@ -50,9 +61,19 @@ def handle_io(self, timeout):
             else:
                 timeout = 0
                 # and loop back to the start
+        return events
+
+    def process_events(self, events):
         for event in events:
             key = (event.ident, event.filter)
-            receiver = self._registered[key]
+            try:
+                receiver = self._registered[key]
+            except KeyError:
```

Comment:
Maybe worth switching this to use the same idiom you used in the epoll manager? (ie, check whether it's the wakeup fd first, rather than catching the KeyError)